src/Sets/Interval.jl

```diff
@@ -633,3 +636,65 @@ function rectify(x::Interval{N}) where {N<:Real}
         return Interval(zero(N), max(x.dat.hi, zero(N)))
     end
 end
+
+"""
+    diameter(x::Interval, [p]::Real=Inf)
+
+Compute the diameter of an interval.
+
+### Input
+
+- `x` -- interval
+- `p` -- (optional, default: `Inf`) norm
+
+### Output
+
+A real number representing the diameter.
+
+### Notes
+
+In one dimension all norms are equivalent.
+"""
+function diameter(x::Interval, p::Real=Inf)
+    return norm(max(x) - min(x), p)
```

Comment:
i've noted that the correct (wrt the floating-point standard) implementation used directed rounding, see here https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/a48c26a837fb7e8e00138cef75064061551ae59d/src/intervals/arithmetic.jl#L494

so i'm wondering if we could add two functions, one `diameter(::Interval)` which just calls IntervalArithmeticl.diam, and another generalized function with two arguments .

Reply:
like

```julia
diameter(x::Interval) = IntervalArithmetic.diam(x)
diameter(x::Interval, p::Real) = norm(max(x) - min(x), p)
```

Reply:
Hm, in that case I would suggest to always use `norm(IntervalArithmetic.diam(x), p)`, unless the interval is empty (in which case I would rather expect the result `0`).

(By the way, I know that the `norm` is redundant for numbers but it costs the same, so I kept it. I thought that maybe there are "invalid" `p`s that you can pass and then it might be preferable to get an error instead of silently ignoring them.)

Reply:
> Hm, in that case I would suggest to always use norm(IntervalArithmetic.diam(x), p), 

ok, makes sense :+1: 

> unless the interval is empty (in which case I would rather expect the result 0).

well since we dont check that intervals are well formed i wouldn't make additional checks  about emptiness in this function...? just solve the construction issue .

```julia
julia> Interval(2, 1)
Interval{Float64,IntervalArithmetic.Interval{Float64}}([2, 1])

julia> diam(Interval(2, 1).dat)
-1.0
```

Reply:
but the call the `norm` is needless; we can check that `p` actually satisfies `1 <= p <= Inf` that's all

Reply:
Oh, I was referring to [this line](https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/a48c26a837fb7e8e00138cef75064061551ae59d/src/intervals/arithmetic.jl#L492) and thought that it would be triggered because I thought that `isempty` would be meant for flat intervals. But it is not :+1: